pyqgiswps/executors/io/filesio.py

```diff
@@ -128,18 +129,35 @@ def parse_output_definition(outdef: QgsProcessingOutputDefinition, kwargs,
     elif isinstance(outdef, QgsProcessingOutputFile):
         # Try to get a corresponding inputFileDefinition
         # See https://qgis.org/pyqgis/master/core/QgsProcessingParameterFileDestination.html
-        mime = None
+        supported_formats = []
         if alg:
             inputdef = alg.parameterDefinition(outdef.name())
             if isinstance(inputdef, QgsProcessingParameterFileDestination):
-                mime = mimetypes.types_map.get("." + inputdef.defaultFileExtension())
+                default_extension = inputdef.defaultFileExtension()
+                default_extension_idx = -1
+                for filter_ in inputdef.fileFilter().split(";;"):
+                    extension_regex = QRegularExpression("[*.]([a-z]+)")
+                    match = extension_regex.match(filter_)
+                    if match.hasMatch():
+                        extension = match.captured(1)
+                        if extension == default_extension:
+                            default_extension_idx = len(supported_formats)
+
+                        supported_formats.append(Format(match.captured(1)))
```

Comment:
You have already assigned `extension` to  `match.captured(1)`, so you may use it as `Format` argument

Reply:
done.